configuration/template/stack.go

```diff
@@ -55,224 +49,112 @@ type ToPtreeFunc func(string, string) string
 type MultiVarConfigAccessFuncs map[string]GetMultiVarConfigFunc
 type GetMultiVarConfigFunc func(string, string) string
 
-func MakeConfigAccessFuncs(confSvc ConfigurationService, varStack map[string]string) ConfigAccessFuncs {
-	return ConfigAccessFuncs{
-		"GetConfigLegacy": func(path string) string {
-			defer utils.TimeTrack(time.Now(), "GetConfigLegacy", log.WithPrefix("template"))
-			query, err := componentcfg.NewQuery(path)
-			if err != nil {
-				return fmt.Sprintf("{\"error\":\"%s\"}", err.Error())
-			}
-
-			payload, err := confSvc.GetComponentConfiguration(query)
-			if err != nil {
-				log.WithError(err).
-					WithField("path", query.Path()).
-					Warn("failed to get component configuration")
-				return fmt.Sprintf("{\"error\":\"%s\"}", err.Error())
-			}
-
-			fields := Fields{WrapPointer(&payload)}
-			err = fields.Execute(confSvc, query.Path(), varStack, nil, make(map[string]texttemplate.Template), nil)
-			log.Warn(varStack)
-			log.Warn(payload)
-			return payload
+func MakeConfigAccessObject(confSvc ConfigurationService, varStack map[string]string) map[string]interface{} {
+	obj := make(map[string]interface{})
+	obj["config"] = map[string]interface{}{
+		"Get": func(path string) string {
+			return getConfig(confSvc, varStack, path)
 		},
-		"GetConfig": func(path string) string {
```

Comment:
Please keep the old names for some time, as these are used in automatically generated templates. Can we have some transition period?
See https://github.com/AliceO2Group/AliceO2/blob/dev/Framework/Core/src/O2ControlHelpers.cxx#L220

Reply:
Yes this is not a problem, the old names will be kept indefinitely but new functions will only be added in the new namespaces.